tests/integration/logic.rs

```diff
@@ -0,0 +1,390 @@
+
+        assert_eq!(3, answers.len());
+
+        assert!(answers.get(0).unwrap().explainables.is_some());
+        assert!(answers.get(1).unwrap().explainables.is_some());
+        assert!(answers.get(2).unwrap().explainables.is_some());
+
+        let age_in_days = transaction.concept().get_attribute_type(String::from("age-in-days")).await?.unwrap();
+        for ans in answers {
+            match ans.map.get("x").unwrap() {
+                Concept::Entity(entity) => {
+                    let attributes: Vec<Attribute> = entity.get_has(&transaction, vec![age_in_days.clone()], vec![])?.try_collect().await?;
+                    if attributes.first().unwrap().value == Value::Long(15) {
+                        assert_single_explainable_explanations(&ans, 1, 1, &transaction).await;
+                    } else {
+                        assert_single_explainable_explanations(&ans, 1, 2, &transaction).await;
+                    }
+                },
+                _ => panic!("Incorrect Concept type: {:?}", ans.map.get("x").unwrap()),
+            }
+        }
+
+        Ok(())
+    }
+}
+
+async fn assert_single_explainable_explanations(
+    ans: &ConceptMap,
+    explainables_count: usize,
+    explanations_count: usize,
+    transaction: &Transaction<'_>,
+) {
+    check_explainable_vars(ans);
+    let explainables = ans.clone().explainables.unwrap();
+    let mut all_explainables = explainables.attributes.values().collect::<Vec<_>>();
+    all_explainables.extend(explainables.relations.values().collect::<Vec<_>>());
+    all_explainables.extend(explainables.ownerships.values().collect::<Vec<_>>());
+    assert_eq!(explainables_count, all_explainables.len());
+    let explainable = all_explainables.get(0).unwrap();
+    assert!(explainable.id >= 0);
+    let stream = transaction.query().explain(explainable.id);
+    assert!(stream.is_ok());
+    let result = stream.unwrap().try_collect::<Vec<_>>().await;
+    assert!(result.is_ok());
+    let explanations = result.unwrap();
+    assert_eq!(explanations_count, explanations.len());
+    for explanation in explanations {
+        let mapping = explanation.variable_mapping;
+        let projected = apply_mapping(&mapping, ans);
+        for var in projected.map.keys() {
+            assert!(explanation.conclusion.map.contains_key(var));
+            assert_eq!(explanation.conclusion.map.get(var), projected.map.get(var));
+        }
+    }
+}
+
+fn check_explainable_vars(ans: &ConceptMap) {
+    ans.clone().explainables.unwrap().relations.into_keys().for_each(|k| assert!(ans.map.contains_key(k.as_str())));
+    ans.clone().explainables.unwrap().attributes.into_keys().for_each(|k| assert!(ans.map.contains_key(k.as_str())));
+    ans.clone()
+        .explainables
+        .unwrap()
+        .ownerships
+        .into_keys()
+        .for_each(|(k1, k2)| assert!(ans.map.contains_key(k1.as_str()) && ans.map.contains_key(k2.as_str())));
+}
```

Comment:
What does this `check`? a better name for the helper would go a long way. Something along the lines of `assert_explainables_in_concept_map`, perhaps?

Reply:
Also, why `clone()`?